src/test/crimson/test_monc.cc

```diff
@@ -35,16 +35,15 @@ static seastar::future<> test_monc()
       if (conf->ms_crc_header) {
         msgr->set_crc_header();
       }
-      return seastar::do_with(MonClient{*msgr},
-                              [msgr](auto& monc) {
-        return msgr->start(&monc).then([&monc] {
-          return seastar::with_timeout(
-            seastar::lowres_clock::now() + std::chrono::seconds{10},
-            monc.start());
-        }).then([&monc] {
-          return monc.stop();
-        });
-      }).finally([msgr] {
+      auto monc = std::make_unique<MonClient>(*msgr);
+      auto monc_ptr = monc.get();
+      return msgr->start(monc_ptr).then([monc_ptr] {
+        return seastar::with_timeout(
+          seastar::lowres_clock::now() + std::chrono::seconds{10},
+          monc_ptr->start());
+      }).then([monc_ptr] {
+        return monc_ptr->stop();
+      }).finally([msgr, monc=std::move(monc)] {
```

Comment:
the reason `monc` is alive is that `monc_ptr` is captured by the `finally` continuation where it is destructed. it works, but it's a little bit difficult to follow. i am adding the move constructor back in hope to improve the readability at #27515 .

could you help review it if it makes sense to you?